### PR TITLE
Various fixes and improvements

### PR DESCRIPTION
**Pull request by ge0rdi**

### CODE REVIEW

**Comment by ge0rdi** [comment]
@bonzibudd
@Gaurav-Original-ClassicShellTester

Guys, appreciate if you could have a look.

**Comment by bonzibudd** [comment]
For me, the taskbar customizations still seem to be broken, including textures. It seems that all of the customizations are happening below the taskbar instead of above. You may know about this, but I couldn't tell from your description.

Whenever I try to do a search, Explorer crashes a few seconds in to loading settings. I guess this has something to do with the ModernSettings changes.

I like the new options for accelerators, I personally think that "Require Alt key for accelerators" should be on by default, since it aligns with Windows behavior, but you may disagree. Also, for clarity, the "Disable Accelerators" option should grey out the "Require Alt key" option, since it doesn't make sense to have both enabled at once.

I also think that the visual indicator for accelerators should be updated to reflect the new options (i.e. when you press a key that isn't an accelerator, it underlines the letter on appropriate items). So the indicator should never show when accelerators are disabled, and only with Alt key when that option is enabled.

Thanks again for your work with this, and let me know if I can help troubleshoot anything.

**Comment by ge0rdi** [comment]
@bonzibudd 
> For me, the taskbar customizations still seem to be broken, including textures.

What's your Windows version?
Can you provide screenshot what's broken exactly?

> Whenever I try to do a search, Explorer crashes a few seconds in to loading settings.

Please, provide dump from Explorer ([How to create dump on Windows Explorer crash](https://github.com/Open-Shell/Open-Shell-Menu/wiki/How-to-create-dump-on-Windows-Explorer-crash)).

> I like the new options for accelerators

Thanks :) I've finally had time (and will) to go through all those accelerator mistake issues and decided to add those options.

Also I think I have fixed the case where it was possible to trigger accelerator before search got focus when displaying start menu.
So hopefully triggering accelerator by mistake should be less frequent now.

> I personally think that "Require Alt key for accelerators" should be on by default, since it aligns with Windows behavior

Yup, I was thinking about that too. I like when things align to _standard_ behavior.
But I'm really not sure how many people are used to current behavior of accelerators. So for now I'd just provide the option and keep the default behavior.
We can change the default later.

> Also, for clarity, the "Disable Accelerators" option should grey out the "Require Alt key" option, since it doesn't make sense to have both enabled at once.

That's a good point 👍 
In such case I'd probably rename `Disable accelerators` to `Enable Accelerators` (by default enabled) and indent the Alt option (so it is clear it is related to the other one).

It could look like this by default:
image
And this when disabled:
image

What do you think?

> So the indicator should never show when accelerators are disabled, and only with Alt key when that option is enabled.

It should work like this when accelerators are disabled. No accelerator cues (those underscores) should be displayed.

I'll try to improve the "Require Alt key" case so that cues will be shown only if Alt was pressed.

**Comment by bonzibudd** [comment]
I am using build 25262.
Here's with taskbar customization disabled:
image
Enabled with default settings (Can barely notice a difference):
image
Opaque red (#FF0000):
image
Taskbar texture: gQPNALP
With texture applied:
image

Due to only really small changes happening to the taskbar, it's hard for me to tell what it's actually doing.

Here's the DMP for search: 
[dmp.zip](https://github.com/Open-Shell/Open-Shell-Menu/files/10290810/dmp.zip)

I think the screenshot you attached for the accelerators would work perfectly.

**Comment by ge0rdi** [comment]
> I am using build 25262

Ah, I forgot you are using Insider builds.
I'll try to have a look at that (I was focusing to standard builds so far).

> Here's the DMP for search

Thanks. I was able to reproduce it.
It was related to VS2022 migration.
Should be fixed now.

> I think the screenshot you attached for the accelerators would work perfectly.

Cool.
I've pushed those changes, so you can test them with new build.

**Comment by ge0rdi** [comment]
It seems there is some issue with `AppVeyor` downloads again (limit reached for whatever reason :( ).
You can get actual build also from here:
https://ci.appveyor.com/project/ge0rdi/open-shell/build/artifacts

File: Src/StartMenu/StartMenuDLL/StartMenuDLL.rc [hunk 1]
```diff
@@ -637,6 +637,10 @@ BEGIN
     IDS_MENU_USERNAME       "User name text"
     IDS_MENU_USERNAME_TIP   "Enter the text you want to see in the user-name portion of the menu (for skins that show the user name)"
     IDS_PIC_COMMAND         "User picture command"
+    IDS_ENABLE_ACCELERATORS "Enable accelerators"
```
Comment (ge0rdi) [inline Src/StartMenu/StartMenuDLL/StartMenuDLL.rc:640]: @bonzibudd 
Do you think these texts are understandable?
Feel free to propose any changes that will make them better.

Comment (bonzibudd) [reply Src/StartMenu/StartMenuDLL/StartMenuDLL.rc:640]: Yes, I think that works. That way it's clear that the alt key option is dependent on the prior.

**Comment by bonzibudd** [comment]
I will try to test the Taskbar changes on a non-insider build soon.

**Comment by bonzibudd** [comment]
Hi @ge0rdi,

I was able to test the taskbar on a non-insider version (22000.1335) and I'm still having some issues. The texture still doesn't fill up the entire area for me, and it glitches in and out often.
Here's what I was first greeted with after enabling the texture:
Screenshot 2022-12-24 231452
Here's after pressing a few taskbar buttons (closer to what it should look like):
Screenshot 2022-12-24 231223
And here's another variation:
Screenshot 2022-12-24 231602

The glitching in and out may be part of another issue, maybe something to do with the program losing focus in favor of other processes. However, the texture still appears to be incomplete.

**Comment by ge0rdi** [comment]
Thanks, I guess I should have a look at 21H2 too (I'm testing with 22H2 as I hope most people will use the newer build eventually).

> It seems more stable on 22621.963, but occasionally a similar issue arises:

Are you able to provide some _steps_ how to reproduce it?

BTW Regarding Insider builds, it seems that the taskbar parts are now drawn by XAML framework not conventional GDI.
And I have no clue so far how to hook/subclass those :(

**Comment by ge0rdi** [comment]
@bonzibudd 
Could you check the taskbar issue with latest build?
I did one improvement.

**Comment by ge0rdi** [comment]
@bonzibudd 
I did some more fixes related to Win11 custom taskbar texture.
Now it works correctly for me on both 22H1 and 22H2.

**Comment by bonzibudd** [comment]
It seems to be fixed on 21h2 now, I will test 22h2 shortly.

**Comment by bonzibudd** [comment]
Looks good on 22h2 as well.

**Review by bonzibudd — APPROVED** [review]

**Pull request merged**